src/components/table/daikin-table.ts

```diff
@@ -0,0 +1,402 @@
+              this._handleClickSort(event, header.key)}
+          >
+            <daikin-table-header-cell
+              alignment=${alignment ?? "left"}
+              ?sortable=${this.sortable && sortable}
+            >
+              ${label}
+            </daikin-table-header-cell>
+          </slot>
+        </th>`;
+      });
+
+    const createRow = (item: T) =>
+      this.headers.map(
+        ({ alignment, ...header }) =>
+          html`<td class="h-full p-0">
+            <slot name=${`cell:${String(header.key)}:${item.id}`}>
+              <daikin-table-cell alignment=${alignment ?? "left"}>
+                ${item[header.key]}
+              </daikin-table-cell>
+            </slot>
+          </td>`
+      );
+
+    return html`<div class="flex flex-col gap-6 w-full font-daikinSerif">
+      <table class="w-full table-fixed leading-[22px]">
+        <thead>
+          <tr class="border-b border-b-system-element-divider-primary">
+            ${this.selectable
+              ? html`<td class="w-12 p-0">
+                  <span
+                    class="flex items-center justify-center w-full h-full min-h-12"
+                  >
+                    <daikin-checkbox
+                      name="allItem"
+                      .checkState=${this._bulkRowsCheckState}
+                      label="Select all"
+                      label-position="hidden"
+                      @change=${this._handleHeaderCheckboxChange}
+                    ></daikin-checkbox>
+                  </span>
+                </td>`
+              : nothing}
+            ${createHeaderRow()}
+          </tr>
+        </thead>
+        <tbody>
+          ${this._currentView.map(
+            (row) =>
+              html`<tr
+                class=${cvaRow({
+                  selected:
+                    this.selectable &&
+                    !!this.selection.find((id) => id === row.id),
+                })}
+              >
+                ${this.selectable
+                  ? html`<td class="w-12 p-0">
+                      <span
+                        class="flex justify-center items-center w-full h-full min-h-12"
+                      >
+                        <daikin-checkbox
+                          name=${row.id}
+                          .checkState=${this.selection.includes(row.id)
+                            ? "checked"
+                            : "unchecked"}
+                          label="Select row"
+                          label-position="hidden"
+                          @change=${this._handleBodyCheckboxChange}
+                        ></daikin-checkbox>
+                      </span>
+                    </td>`
+                  : nothing}
+                ${createRow(row)}
+              </tr>`
+          )}
+        </tbody>
+      </table>
+    </div>`;
+  }
+
+  protected override willUpdate(): void {
+    this._updateTable();
+    this._updateHeaderCheckboxState();
+  }
```

Comment:
Performance improvement.

```suggestion
  protected override willUpdate(changedProperties: PropertyValues<this>): void {
    const viewChanged = changedProperties.has("rows") || changedProperties.has("sort") || changedProperties.has("order") || changedProperties.has("sortFunction");
    const selectionChanged = changedProperties.has("selection");

    if (viewChanged) {
      this._updateCurrentView();
    }
    if (viewChanged || selectionChanged) {
      this._updateHeaderCheckboxState();
    }
  }
```